maths/makevalid/main.go

```diff
@@ -65,7 +63,33 @@ func destructure2(polygons [][]maths.Line, clipbox *points.BoundingBox) []maths.
 	return segments
 }
 
-func destructure5(ctx context.Context, hm hitmap.Interface, clipbox *points.BoundingBox, segments []maths.Line) ([][][]maths.Pt, error) {
+func MaxF64(vals ...float64) (max float64) {
```

Comment:
This is a copy of the math.Max method made so that it is faster (by removing extra code.) I don't need to check to see if the value is NAN or Inf, as these are things I can assume from the construction of the function. Whereas math.Max is more generic and has to deal with overflows and special values.